File: OOPs/_3_Inheritance/inheritance.py
class Animal:
    def __init__(self, name):
        self.name = name
    
    def make_sound(self):
        print(f'{self.name} makes a sound')

class Dog(Animal):
    def make_sound(self):
        print(f'{self.name} barks')

dog = Dog('Dog')
dog.make_sound()  # Dog barks

File: OOPs/_3_Inheritance/inheritance.dart
class Animal {
  String name;
  
  Animal(this.name);
  
  void makeSound() {
    print('$name makes a sound');
  }
}

class Dog extends Animal {
  Dog(String name) : super(name);
  
  @override
  void makeSound() {
    print('$name barks');
  }
}

void main() {
  Dog dog = Dog('Dog');
  dog.makeSound();  // Dog barks
}
